vespalib/src/vespa/vespalib/datastore/sharded_hash_map.h

```diff
@@ -56,6 +56,9 @@ class ShardedHashMap {
     size_t size() const noexcept;
     const EntryComparator &get_default_comparator() const noexcept { return *_comp; }
     MemoryUsage get_memory_usage() const;
+    void foreach_key(std::function<void(EntryRef)> callback) const;
```

Comment:
Consider adding unit tests for these new functions (foreach_key, move_keys, and normalize_values). I guess it makes most sense putting them in sharded_hash_map_test.cpp.